code/game/machinery/firealarm.dm

```diff
@@ -143,16 +155,38 @@
 		return
 	var/area/A = get_area(src)
 	A.firereset(src)
+	update_appearance()
 	if(user)
 		log_game("[user] reset a fire alarm at [COORD(src)]")
 
+/obj/machinery/firealarm/proc/try_lock(mob/user, force_lock = FALSE)
+	if(allowed(user) || !user || force_lock)
+		if(!locked || force_lock)
+			locked = TRUE
+			balloon_alert(user, "Locked")
+		else
+			locked = FALSE
+			balloon_alert(user, "Unlocked")
+		playsound(src, 'sound/machines/beep.ogg', 50, 1)
+	else
+		balloon_alert(user, "Access Denied!")
+		playsound(src, 'sound/machines/terminal_error.ogg', 50, 1)
+	update_appearance()
+
+/obj/machinery/firealarm/AltClick(mob/user)
```

Comment:
Would prefer if the interaction was performed by clicking on it with an item rather than on alt click due to alt click interfering with the atom panel, along with alt click becoming less common as an interaction.

Reply:
Why not both? Roboticists can use ALT click and ID to lock and unlock Cyborg panels.